### DISCUSSION TITLE
Workbench for ory/kratos node/javascript/typescript SDK tests with Angular

### DISCUSSION THREAD

**Discussion by TMInnovations** [ory/kratos#915] [Ideas]
First: **Big thanks to the ory team for starting such a great project!**

I tried to integrate the ory/kratos node/javascript/typescript sdk into one of my projects (frontend) and discovered some troubles. To better understand these I spun up a minimal git repo, that can be found [here](https://github.com/TMInnovations/testorykratos). Feel free to play with it and evaluate the sdk. 

I'd like to extend this repo if there's interest from other sides and am very thankful for additional contributors.

For one problem a question on stackoverflow already has been asked and was discussed a bit. You will find it  [here](https://stackoverflow.com/q/65319231/14345380). No solution has been found yet. You are welcome to join the discussion.

**Comment by vinckr** [comment]
Thanks for pointing this out!

I actually have an issue assigned to update the Js/Ts Sdk documentation: https://github.com/ory/hydra/pull/2195, 
so maybe your problem could serve to improve the documentation there somewhat.
Will check out your repo later and try to reproduce the issues.
I have not really used the sdk before, but I see you have put some instruction on your repo, very nice!

**Reply by TMInnovations** [reply]
I'd love to contribute to the documentation of the sdk as soon as I entirely understand the problems I already faced with it. 

For SPA it's important to not have to leave the page to register/login/... because the variables set inside the app are gone once the reload happened. I think the problem can be solved by giving the sdk user a chance to opt out of the (browser) redirection.

I think that developing SPAs with the sdk as it is now is not possible due to https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSExternalRedirectNotAllowed
